packages/astro/src/server/clerk-middleware.ts

```diff
@@ -94,24 +103,39 @@ export const clerkMiddleware: ClerkMiddleware = (...args: unknown[]): any => {
       throw new Error('Clerk: handshake status without redirect');
     }
 
-    const authObject = requestState.toAuth();
+    const authObjectFn = (opts?: PendingSessionOptions) => requestState.toAuth(opts);
 
     const redirectToSignIn = createMiddlewareRedirectToSignIn(clerkRequest);
-    const authObjWithMethods: ClerkMiddlewareAuthObject = Object.assign(authObject, { redirectToSignIn });
 
-    decorateAstroLocal(clerkRequest, context, requestState);
+    decorateAstroLocal(clerkRequest, authObjectFn, context, requestState);
 
     /**
      * ALS is crucial for guaranteeing SSR in UI frameworks like React.
      * This currently powers the `useAuth()` React hook and any other hook or Component that depends on it.
      */
-    return authAsyncStorage.run(context.locals.auth(), async () => {
+    const asyncStorageAuthObject =
+      authObjectFn().tokenType === TokenType.SessionToken ? authObjectFn() : signedOutAuthObject({});
```

Comment:
For ALS, we dont really need the `redirectToSignIn` method to be passed to `useAuth()`